src/SFA.DAS.ProviderRelationships/Exceptions/MissingEntityException.cs

```
@@ -0,0 +1,12 @@
+using System;
+
+namespace SFA.DAS.ProviderRelationships.Exceptions
+{
+    public class MissingEntityException/*<TEntity>*/ : Exception
```

Could this class be removed? If we use `Single()` it will throw for us without needing our own null check and throw.

---

If we use `Single()` it would probably make sense to remove the null check and this exception, but if we continue to use `FindAsync()` (see comment about `FindAsync()` vs `SingleAsync()`), it is cleaner to throw this exception, rather than letting it fall through to a `NullReferenceException`.

---

Same comment here as https://github.com/SkillsFundingAgency/das-provider-relationships/pull/19#discussion_r227732299

---

As we've switched to SingleAsync(), I've removed this exception.